components/content-service/pkg/layer/provider.go

```diff
@@ -368,6 +508,17 @@ func contentDescriptorToLayer(cdesc []byte) (*Layer, error) {
 	)
 }
 
+// version of this function for persistent volume claim feature
+// we cannot use /workspace folder as when mounting /workspace folder through PVC
+// it will mask anything that was in container layer, hence we are using /.workspace instead here
+func contentDescriptorToLayerPVC(cdesc []byte) (*Layer, error) {
+	return layerFromContent(
+		fileInLayer{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
+		fileInLayer{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace/.gitpod", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
+		fileInLayer{&tar.Header{Typeflag: tar.TypeReg, Name: "/.workspace/.gitpod/content.json", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755, Size: int64(len(cdesc))}, cdesc},
+	)
+}
```

Comment:
Considering that we're not using FWB anywhere, we could harmonise this path and make `/.workspace` the default for those files. Removes the duplication herein and and in supervisor. Same goes for `workspaceReadyLayerPVC`.

Reply:
I agree! I thought of doing that change, but was concerned about some unknown side effect to regular workspaces, so instead concentrated on adding PVC path that would not affect normal workflow right now. 
I added an issue to make this happen as part of separate PR though: https://github.com/gitpod-io/gitpod/issues/9496